File: app/src/main/java/com/example/tribecrypto/database/currency/CurrencyDao.java
package com.example.tribecrypto.database.currency;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.example.tribecrypto.data.entity.CryptoCurrencyEntity;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Single;

@Dao
public interface CurrencyDao {

    @Query("SELECT * FROM crypto_currency")
    Observable<List<CryptoCurrencyEntity>> getAllCurrencies();

    @Query("SELECT * FROM crypto_currency WHERE name LIKE :name")
    Single<CryptoCurrencyEntity> getEntityByName(String name);

    @Update
    void updateAll(List<CryptoCurrencyEntity> data);

    @Update(entity = CryptoCurrencyEntity.class)
    void update(CryptoCurrencyEntity date);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertAll(List<CryptoCurrencyEntity> data);

    @Query("DELETE FROM crypto_currency")
    void deleteAllCryptoCurrencies();
}

File: app/src/main/java/com/example/tribecrypto/database/CurrencyDao.kt
package com.example.tribecrypto.database

import androidx.room.*
import com.example.tribecrypto.data.entity.CryptoCurrencyEntity
import io.reactivex.Observable
import io.reactivex.Single

@Dao
interface CurrencyDao {

    @Query("SELECT * FROM crypto_currency")
    fun getAllCurrencies() : Observable<List<CryptoCurrencyEntity>>

    @Query("SELECT * FROM crypto_currency WHERE name LIKE :name")
    fun getEntityByName(name: String): Single<CryptoCurrencyEntity>

    @Update
    fun updateAll(data : List<CryptoCurrencyEntity>)

    @Update(entity = CryptoCurrencyEntity::class)
    fun update(date: CryptoCurrencyEntity)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertAll(data : List<CryptoCurrencyEntity>)

    @Query("DELETE FROM crypto_currency")
    fun deleteAllCryptoCurrencies()
}